### ISSUE TITLE
no decode delegate for this image format `' @ error/constitute.c/ReadImage/501.

### ISSUE THREAD

**Issue by mchun** [flovilmart/parse-image#7]
Thanks for creating this repo.
But I wonder if any delegate of imagemagick has to be installed?
I encountered the caption error when executing setData.

```
Parse.Cloud.httpRequest({ url: url }).then(function(response) {
    // Create an Image from the data.
    var image = new Image();
    return image.setData(response.buffer);
);
```

**Comment by flovilmart** [comment]
@mchun what platform/system are you running on?

**Comment by mchun** [comment]
mac osx 10.10.5
when I do "convert -list configure"
It contains the following, I have no idea what went wrong and currently using jimp instead.
DELEGATES      bzlib mpeg freetype jng jpeg lzma png tiff webp xml zlib

**Comment by flovilmart** [comment]
I'm on 10.11.3 and that works like a breeze, on travis the tests passes too.. that's really odd. Seems a mis installation of the couple gm/imagemagick

**Comment by carmenlau** [comment]
This is encoding problem. Parse.Cloud.httpRequest is using [request](https://github.com/request/request).

> encoding - Encoding to be used on setEncoding of response data. If null, the body is returned as a Buffer. Anything else (including the default value of undefined) will be passed as the encoding parameter to toString() (meaning this is effectively utf8 by default). (Note: if you expect binary data, you should set encoding: null.)

Set encoding to null can solve the problem for me.

```
Parse.Cloud.httpRequest({ url: url, encoding: null}).then(function(response) {
    // Create an Image from the data.
    var image = new Image();
    return image.setData(response.buffer);
);
```

**Comment by flovilmart** [comment]
Good catch! Do you have time for a Pr?

**Comment by carmenlau** [comment]
Saw your PR in parse-server for this issues. Thanks a lot and appreciate for your work!

**Comment by tsanthosh** [comment]
Hi @flovilmart, Thanks for doing this. I am getting the same error too when trying to download an image from a url using httpRequest. I even tried using `encoding = null` but still doesn't work. Can you please let me know how I should work around this error.

**Comment by flovilmart** [comment]
can you try without using Parse.Cloud.httpRequest but a simple http request manager and see if it works passing the response buffer to ParseImage

**Comment by mcclaskiem** [comment]
Ive tried two different request clients here and am still getting the same error. Is this something that is occuring across all file adapters or could it be provider specific?

**Comment by flovilmart** [comment]
I'm not sure. where is it deployed? heroku?

**Comment by mcclaskiem** [comment]
Im running it locally on 10.10.5

**Comment by flovilmart** [comment]
uhm, never had that issue on mac...

**Comment by mcclaskiem** [comment]
both graphmagick and imagemagick are installed via homebrew, is there something I can check to confirm with `confirm -list configure`

**Comment by flovilmart** [comment]
Can you try cloning the repo and run the tests? With npm run tests?

**Comment by mcclaskiem** [comment]
Got 8 passing here, must be something with the response that I am passing.

**Comment by flovilmart** [comment]
yup that's most likely, make sure that's a proper buffer with an image in it :)

**Comment by mcclaskiem** [comment]
Where do you suggest posting to verify a FileAdapter? I have one written for RackSpace but would like to make sure its conformant to parse-server

**Comment by flovilmart** [comment]
try locally with your code to save the buffer as a file.
